### PR TITLE
enable push/pull only when it has a published branch

### LINKED ISSUE #1448: Grey out Repository menu items for New Repo
If user creates a new repo (File > New Repo), without publishing, then there should be some Repository menu items that are greyed out as they aren't relevant. 

For instance, "View on GitHub" and "Pull". Clicking "Pull" results in terminal error with no message surfaced to the user:

screen shot 2017-05-12 at 9 25 18 am

v0.5.0, OSX 10.11.5

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #1448

### CODE REVIEW

File: app/src/lib/menu-update.ts [hunk 1]
```diff
@@ -145,8 +145,8 @@ function getMenuState(state: IAppState): Map<MenuIDs, IMenuItemState> {
     menuStateBuilder.setEnabled('compare-branch', isHostedOnGitHub && hasPublishedBranch)
 
     menuStateBuilder.setEnabled('view-repository-on-github', isHostedOnGitHub)
-    menuStateBuilder.setEnabled('push', !networkActionInProgress)
-    menuStateBuilder.setEnabled('pull', !networkActionInProgress)
+    menuStateBuilder.setEnabled('push', hasPublishedBranch && !networkActionInProgress)
```
Comment (joshaber) [inline app/src/lib/menu-update.ts]: This doesn't seem quite right... if the branch hasn't been published yet, they should be able to push it to publish it.

Comment (shiftkey) [inline app/src/lib/menu-update.ts]: This is a good shout. This is how it is as of this PR:

 - for an unpublished repository, `Publish Repository` will also publish the `master` branch automagically
 - for a new branch after the repository has been published, you'll see this:



We don't look to see if any remotes have been defined for the repository - does that seem like a more relaxed approach here?

Comment (shiftkey) [inline app/src/lib/menu-update.ts]: #jolo

**Comment by shiftkey** [comment]
:moon:

**Pull request merged**